opentracing-api/src/main/java/io/opentracing/Tracer.java

```diff
@@ -20,73 +20,80 @@
 public interface Tracer {
 
   /**
-   * Create, start, and return a new Span with the given `operationName`.
-   * An optional parent Span can be specified used to incorporate the newly-returned Span into an existing trace.
+   * Return a new SpanBuilder for a Span with the given `operationName`.
    *
-   * <p>Example:
+   * <p>A contrived example:
    * <pre>{@code
     Tracer tracer = ...
 
-    Span feed = tracer.buildSpan("GetFeed")
-                      .start();
+    Span parentSpan = tracer.buildSpan("DoWork")
+                            .start();
 
     Span http = tracer.buildSpan("HandleHTTPRequest")
-                      .withParent(feed)
+                      .withChildOf(parentSpan.context())
                       .withTag("user_agent", req.UserAgent)
                       .withTag("lucky_number", 42)
                       .start();
     }</pre>
    */
   SpanBuilder buildSpan(String operationName);
 
-  /** Takes two arguments:
-   *    a Span instance, and
-   *    a “carrier” object in which to inject that Span for cross-process propagation.
+  /**
+   * Inject a SpanContext into a `carrier` of a given `format`, presumably for propagation across process boundaries.
+   *
+   * <p>Example:
+   * <pre>{@code
+   * Tracer tracer = ...
+   * Span clientSpan = ...
+   * TextMapWriter httpHeaderWriter = new AnHttpHeaderCarrier(httpRequest);
+   * tracer.inject(span.context(), BuiltinFormats.HTTP_HEADER, httpHeaderWriter);
+   * }</pre>
    *
-   * A “carrier” object is some sort of http or rpc envelope, for example HeaderGroup (from Apache HttpComponents).
+   * @param spanContext the SpanContext instance to inject into the carrier
+   * @param format the Format of the carrier. See BuiltinFormats.
+   * @param carrier the carrier for the SpanContext state; when inject() returns, the Tracer implementation will have represented the SpanContext within `carrier`
    *
-   * Attempting to inject to a carrier that has been registered/configured to this Tracer will result in a
-   * IllegalStateException.
+   * All implementations must support, at minimum, the BuiltinFormats.
    *
-   * All implementations support at minimum the required carriers BinaryWriter and TextMapWriter.
+   * @see BuiltinFormats
    */
-  <T> void inject(Span span, T carrier);
+  <W> void inject(SpanContext spanContext, Format<?, W> format, W carrier);
```

Comment:
Users, exactly. Right now the users see this method:

```java
void <T> inject(s SpanContext, T carrier)
``` 

Since `T` is unconstrained in any way, it's equivalent to 

```java
void inject(s SpanContext, Object carrier)
``` 

As a user, what am I supposed to pass as `carrier`?

Now consider the case with Format:

```java
void <W> inject(s SpanContext, Format<?, W> format, W carrier)
``` 

What would the user do? They will look at Format class and see a constant: 

```java
Format<.., TextMapWriter> TextMap = ...
``

Now they know what to pass as the `format` parameter, and it explicitly constraints what they should pass as a `carrier`:

```java
tracer.inject(span.getContext(), Format.TextMap, new TextMapWriter(){...})
```